### DISCUSSION TITLE
Refresh cached fetch data by key

### DISCUSSION THREAD

**Discussion by Gyurmatag** [nuxt/nuxt#17605] [Questions] [answered]
So I've got a method here:

```
const { data } =
  await useCustomFetch<ReservationList>({
    path: `${RESERVATION}/${USER_RESERVATIONS}`,
    isAuthenticated: true,
  });

userStore.setReservations(data.value.reservations);
```

This is calling a useCustomFetch method which is a useFetch method. It is listing reservations on page load and storing it into the store. In this same component I've got a delete method too:

```
const deleteReservation = async (reservationId) => {
  const { data } = await useCustomFetch({
    path: `${RESERVATION}/${reservationId}`,
    method: FetchMethods.DELETE,
    isAuthenticated: true,
  });

  if (data) {
    userStore.deleteReservation(reservationId);
    // I want to delete / refresh the cached reservation list here
  }
}; 
```

I want to use a key like mentioned in the docs here: https://v3.nuxtjs.org/guide/features/data-fetching
But when using useFetch the keys are auto-generated. How will I know the key or is there any easyer way? Maybe specify a custom key using useFetch?

Thanks for the help!

**Comment by gvalex13** [comment]
It seems the only way to control using cache is to make own wrapper composable for useAsyncData. I need to update the data every time when a user visits the page. And I did not find other decision

**Comment by hacknug** [comment, marked as answer]
`useFetch` also returns a `refresh` method which you can use to get the new data: https://v3.nuxtjs.org/api/composables/use-fetch#return-values

**Reply by gvalex13** [reply]
It's possible to use it after you made the first call opening the page. But when the visitor goes from one page to another, it's useless. The first call already uses the cache

**Reply by gvalex13** [reply]
But there is an option for useAsyncData to ignore the cache, so I think that's the solution

**Reply by Gyurmatag** [reply]
Yeah it's working but unfortunately there is a new problem described here: https://github.com/nuxt/framework/discussions/4869
Have you got any idea why is that problem happening?
